### PR TITLE
OffScreenMargin calculation with different DPI monitors calculation

### PR DESCRIPTION
**Pull request by ltetak**
## What does the pull request do?
Fixes invalid window margin on systems with two monitors with different DPI scaling (e.g. notebook with an external monitor).

## What is the current behavior?
the border is calculated with the DPI scaling of the actual monitor. But the AdjustWindowRectEx method is not DPI aware so we need to calculate it with the DPI of the Primary screen.
**Repro steps:**
1. you need two monitors with different DPI scalings
2. start ControlCatalog demo on the **secondary** screen
3. set ExtendedClientAreaToDecorations=true
image
4. Maximize
image
5. you can see a white border around the app (can be transparent or a different color depending on other settings)
image

## What is the updated/expected behavior with this PR?
Border should be correct

## How was the solution implemented (if it's not obvious)?
Maybe there is a better way to get the primary screen DPI. Feel free to improve.
I also believe there are other places where this does not produce correct value. I fixed just this one where I can reproduce the problem.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
Should be ok for both monitor-aware DPI scaling and non-aware because the windows API is non-aware.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021239-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/WindowImpl.cs [hunk 1]
```diff
@@ -208,6 +209,8 @@ public Thickness BorderThickness
             }
         }
 
+        public double PrimaryScreenRenderScaling => Screen.AllScreens.FirstOrDefault(screen => screen.Primary)?.PixelDensity ?? 1;
```
Comment (jmacato) [inline src/Windows/Avalonia.Win32/WindowImpl.cs:212]: I wonder if this should be window specific? not just the first monitor? @kekekeks @maxkatz6

**Review by jmacato — DISMISSED** [review]

File: src/Windows/Avalonia.Win32/WindowImpl.cs (hunk 1 above)
Comment (maxkatz6) [reply src/Windows/Avalonia.Win32/WindowImpl.cs:212]: It seems to be intended "the border is calculated with the DPI scaling of the actual monitor. But the AdjustWindowRectEx method **is not DPI aware so we need to calculate it with the DPI of the Primary screen**."

Comment (danwalmsley) [inline src/Windows/Avalonia.Win32/WindowImpl.cs:212]: this can be private I think?

**Review by danwalmsley — DISMISSED** [review]

File: src/Windows/Avalonia.Win32/WindowImpl.cs (hunk 1 above)
Comment (ltetak) [reply src/Windows/Avalonia.Win32/WindowImpl.cs]: @danwalmsley yes, although it is private property then. Changed.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021361-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
@ltetak wouldn't it be better to use AdjustWindowRectExForDpi instead?

**Comment by ltetak** [comment]
@maxkatz6 yes and no. All AdjustWindowRectExForDpi adds is an extra "dpi" argument. I believe it just internally multiplies everything by some ratio (96/dpi or something like that) and it still does not account for multiple monitors.
So we could move this division/multiplication away from our logic but this would also make it different from other cases where it is still calculated on Avalonia side. But I am not the one to decide. Feel free to rewrite in any way. Thanks.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021409-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by GMas0124816** [comment]
The same problem is in place if the monitors are the same size, but the taskbar is only on the main screen. I have the title bar removed and use a custom bar, but the same is in place with the title bar 'on'. In 'full screen' and 'normal' mode all is perfect , but when maximized an offset of a number of pixels (between 5 and 10 is my guess) exists. If I start the app in 'full screen' mode and afterwards go to normal and then to maximized, I can see the offset on the other monitors. If I start in maximized mode and then go to normal and maximize again, the offset is transparent (do not see it on the other monitors).

Do not know if this is solved in the 0.10.999 version as well ? Currently I'm using the 'EffectiveViewportChanged' event to offset it correctly myself for the title bar (on my monitors). The only problem I have, since it is a dpi matter, is if my offset will be acceptable on all types of monitors ...

**Comment by ltetak** [comment]
@GMas0124816 it is not the size that matters but the DPI scaling. Do both your monitors have the same DPI scaling?

**Comment by GMas0124816** [comment]
Yes, it is tree exact same monitors (same DPI scaling, same model even successive serial numbers) where the middle one is the main screen. It is one Win11 pro, latest 0.10.15 avalonia version.

**Pull request merged**